website/src/hooks/ui/useFallbackRef.ts

```diff
@@ -0,0 +1,7 @@
+import { ForwardedRef, RefObject, useRef } from "react";
+
+export const useFallbackRef = <T>(maybeRef?: ForwardedRef<T>) => {
+  const ref = useRef(null);
+
+  return (maybeRef || ref) as RefObject<T>;
+};
```

Comment:
I am not sure I understand what this does?

Reply:
@AbdBarho you can read more about it here https://mtsknn.fi/blog/react-fallback-ref